mm2src/mm2_db/src/indexed_db/indexed_cursor.rs

```diff
@@ -1064,4 +1064,49 @@ mod tests {
         // maybe_swap should return swap with uuid4 since it's the item with the highest rel_coin_value in the store.
         assert_eq!(maybe_swap, Some(swap_item!("uuid3", "RICK", "XYZ", 7, u32::MAX, 1281)));
     }
+
+    #[wasm_bindgen_test]
+    async fn test_cursor_where_first_condition_with_limit() {
+        const DB_NAME: &str = "TEST_REV_ITER_SINGLE_KEY_BOUND_CURSOR";
+        const DB_VERSION: u32 = 1;
+
+        register_wasm_log();
+
+        let items = vec![
+            swap_item!("uuid1", "RICK", "MORTY", 10, 3, 700),
+            swap_item!("uuid2", "MORTY", "KMD", 95000, 1, 721),
+            swap_item!("uuid3", "RICK", "XYZ", 7, u32::MAX, 1281), // +
+            swap_item!("uuid4", "RICK", "MORTY", 8, 6, 92),        // +
```

Comment:
I think we discussed the problem if there are more than one value where the condition applies and how to do that in DM. I am leaving this comment here to not forget about it and so that you can add it to the related issue as you said it should be done in a separate PR with other enhancements.

Reply:
noted.